prov/shm/src/smr_attr.c

```diff
@@ -32,8 +32,13 @@
 
 #include "smr.h"
 
+#define SMR_TX_CAPS (FI_MSG | FI_SEND | FI_RMA | FI_READ | 		\
+		     FI_WRITE | FI_ATOMICS)
+#define SMR_RX_CAPS (FI_MSG | FI_RECV | FI_RMA | FI_REMOTE_READ |	\
+		     FI_REMOTE_WRITE | FI_ATOMICS | FI_SOURCE)
```

Comment:
I'll wager that most providers are setting the Tx/Rx caps incorrectly when it comes to FI_READ, FI_WRITE, FI_REMOTE_READ, FI_REMOTE_WRITE.  I wonder if the above two defines shouldn't be moved into common code. Maybe something like OFI_BASE_TX_CAPS?  Or maybe break it out more, e.g. OFI_TX_MSG_CAPS, OFI_TX_RMA_CAPS, etc.